### ISSUE TITLE
Sails blueprint GET call by id does not work

### ISSUE THREAD

**Issue by zekokhan** [balderdashy/sails-mongo#370]
I have a simple sails.js project set up with sails-mongo. When I do a sails GET for /user I get all the users in the db BUT when i try to call it with  /user/{id} I do not get anything back. What can I do to fix this? My model is pretty basic: 
```javascript
 attributes: {
    id:{
        type: 'string',
      primaryKey: true
    },
    username: {
      type: 'string',
      unique: true
    },
    email: {
      type: 'email',
      unique: true
    }
}
```
I tried using sails-mongo ObjectId(id) to findOne user by Id on a my own custom sails route to no avail. Please help!

**Comment by zekokhan** [comment]
I got it to work with the find and findone sails models calls. I had to change the id attribute to use type: 'integer': 

```
sails.models['user'].findOne({ where: {  id: userId } }).exec(function(err, results){ ..... }

 attributes: {
    id:{
        type: 'integer',
      primaryKey: true
    },
    username: {
      type: 'string',
      unique: true
    },
    email: {
      type: 'email',
      unique: true
    }
}
```

but the blueprint calls still so not work. Thanks!

**Comment by devinivy** [comment]
You can remove the `id` attribute from your model definition altogether.  If you're using the most recent version of sails-mongo, then the `id` attribute is created automatically and references the underlying `_id` "column" in your mongo collection.

**Comment by sailsbot** [comment]
Thanks for posting, @zekokhan.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
